src/tests/system/tests/test_ldap.py

```diff
@@ -599,3 +599,391 @@ def test_ldap__reset_cached_timestamps_to_reflect_changes(client: Client, ldap:
+        1. Look up each user.
+        2. Verify their shells.
+    :expectedresults:
+        1. User with valid shell keeps it.
+        2. User with invalid shell gets fallback.
+        3. User with empty shell gets default shell.
+    :customerscenario: True
+    """
+    ldap.user("user_with_sh_shell").add(shell="/bin/sh")
+    ldap.user("user_with_not_installed_shell").add(shell="/bin/not_installed")
+    ldap.user("user_with_empty_shell").add(shell="")
+
+    client.sssd.nss["default_shell"] = "/bin/default"
+    client.sssd.nss["allowed_shells"] = "/bin/default, /bin/not_installed"
+    client.sssd.nss["shell_fallback"] = "/bin/fallback"
+    client.sssd.start()
+
+    user_sh = client.tools.getent.passwd("user_with_sh_shell")
+    assert user_sh is not None and user_sh.shell == "/bin/sh"
+
+    user_not_installed = client.tools.getent.passwd("user_with_not_installed_shell")
+    assert user_not_installed is not None and user_not_installed.shell == "/bin/fallback"
+
+    ## TODO: fix
+    user_empty = client.tools.getent.passwd("user_with_empty_shell")
+    assert user_empty is not None and user_empty.shell == "/bin/default"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__nss_vetoed_shells(client: Client, ldap: LDAP):
+    """
+    :title: Test the effect of the "vetoed_shells" option
+    :description:
+        This test checks that the 'vetoed_shells' option in the [nss]
+        section of sssd.conf prevents users from having a shell from the
+        vetoed list.
+    :setup:
+        1. Create users in LDAP with different shells.
+        2. Configure SSSD with 'vetoed_shells', 'default_shell', and
+           'shell_fallback'.
+        3. Start SSSD.
+    :steps:
+        1. Look up each user.
+        2. Verify their shells.
+    :expectedresults:
+        1. User with a valid, non-vetoed shell keeps it.
+        2. User with a vetoed shell gets the fallback shell.
+        3. User with an empty shell gets the default shell.
+    :customerscenario: True
+    """
+    ldap.user("user_with_sh_shell").add(shell="/bin/sh")
+    ldap.user("user_with_vetoed_shell").add(shell="/bin/vetoed")
+    ldap.user("user_with_empty_shell").add(shell="")
+
+    client.sssd.nss["default_shell"] = "/bin/default"
+    client.sssd.nss["vetoed_shells"] = "/bin/vetoed"
+    client.sssd.nss["shell_fallback"] = "/bin/fallback"
+    client.sssd.start()
+
+    user_sh = client.tools.getent.passwd("user_with_sh_shell")
+    assert user_sh is not None and user_sh.shell == "/bin/sh"
+
+    user_vetoed = client.tools.getent.passwd("user_with_vetoed_shell")
+    assert user_vetoed is not None and user_vetoed.shell == "/bin/fallback"
+
+    user_empty = client.tools.getent.passwd("user_with_empty_shell")
+    assert user_empty is not None and user_empty.shell == "/bin/default"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_ldap__nss_filters(client: Client, ldap: LDAP):
```

Comment:
I'd break this up into two tests, users and groups. I think all of these tests can should be generic provider tests. 

To test both schemas, rfc2307 and rfc2307bis, rfc2307 is generally covered by genericProvider tests. Then that test is cloned to test_ldap.py only testing rfc2307bis. The questions we need to answer is, should nss work with all providers. If yes, test_authentication.py or maybe test_nss.py and we can move the override_homedir to test_nss.py.